### PR TITLE
Created all FETCH and POST endpoints

### PR DESCRIPTION
**Pull request by Lam7150**
## Status: DONE
:rocket: Ready
## Description :star2: This PR contains all the endpoints for the project (fetch and post)

Creates FETCH endpoints #6 / #15, #4, and #9 and POST endpoints: #14

### CODE REVIEW

File: src/utils/Firebase/firebase.js [hunk 1]
```diff
@@ -36,23 +31,83 @@ class Firebase {
...
+        );
+        return new Promise((resolve, reject) => {
+          if (deploymentDoc) {
+            resolve(lastMasteredLesson);
+          } else {
+            reject(Error("Deployment doc does not exist"));
+          }
+        });
+      })
+      .catch(error => console.log("Error getting student account: ", error));
```
Comment (chyku) [inline src/utils/Firebase/firebase.js]: something that makes me unsure about the way you're writing endpoint is that Jason doesn't format his endpoints this way and we don't want to deviate too much from the way they do things

File: src/components/ExampleComponent/ExampleComponent.js [hunk 2]
```diff
@@ -1,22 +1,14 @@
...
-      .get()
-      .then(snapshot =>
-        snapshot.docs.map(d => {
-          const data = d.data();
-          return { id: d.id, ...data };
-        })
-      );
-  getData().then(result => console.log(result));
-
-  return <div> Example Component </div>;
```
Comment (chyku) [inline src/components/ExampleComponent/ExampleComponent.js]: can you replace this with an actual example of calling an endpoint you wrote?

File: src/utils/Firebase/firebase.js [hunk 3]
```diff
@@ -36,23 +31,153 @@ class Firebase {
...
-  getLastMastered = deploymentAccountId =>
+  //Gets the most recent lesson that a student has completed
+  getLastMasteredLesson = deploymentAccountId =>
     this.db
       .doc(`deployment_account/${deploymentAccountId}/`)
       .get()
-      .then(documentSnapshot => {
-        documentSnapshot.get("profile.glenLearn.lastMasteredLesson");
-      });
+      .then(deploymentDoc => {
```
Comment (chyku) [inline src/utils/Firebase/firebase.js]: maybe rename this to deploymentAccountDoc to differentiate between deployments and deployment accounts

File: src/utils/Firebase/firebase.js [hunk 4]
```diff
@@ -36,23 +31,153 @@ class Firebase {
...
+          } else {
+            reject(Error("Deployment doc does not exist"));
+          }
+        });
+      })
+      .catch(error => console.log("Error getting student account: ", error));
 
-  //Data access APIs
-  customLessons = adminAccountId =>
+  getTeacherCustomLessons = adminAccountId =>
```
Comment (chyku) [inline src/utils/Firebase/firebase.js]: use "admin" instead of teacher in the name of the endpoint plz

File: src/utils/Firebase/firebase.js [hunk 5]
```diff
@@ -36,23 +31,153 @@ class Firebase {
...
+          if (querySnapshot) {
+            resolve(customLessons);
+          } else {
+            reject(Error("No custom lessons found."));
+          }
+        });
+      })
+      .catch(error => console.log("Error getting custom lessons: ", error));
+
+  getStudentCustomLessons = deploymentAccountId =>
```
Comment (chyku) [inline src/utils/Firebase/firebase.js]: also instead of student use deploymentaccount

File: src/utils/Firebase/firebase.js [hunk 6]
```diff
@@ -36,23 +31,153 @@ class Firebase {
...
+    this.db
+      .doc(`deployment_account/${deploymentAccountId}/`)
+      .get()
+      .then(deploymentDoc => {
```
Comment (chyku) [inline src/utils/Firebase/firebase.js]: same comment as above for deploymentdoc (deploymentacc vs deployment)

File: src/utils/Firebase/firebase.js [hunk 7]
```diff
@@ -36,23 +31,153 @@ class Firebase {
...
+        customLessons: app.firestore.FieldValue.arrayUnion(customLessonRef.id)
+      });
+    }
+
+    return batch
+      .commit()
+      .catch(error =>
+        console.log("Pushing custom lessons to deployments failed: ", error)
+      );
+    // needs better promise rejection
```
Comment (chyku) [inline src/utils/Firebase/firebase.js:179]: i hope u fix these after mvp

**Review by chyku — CHANGES_REQUESTED** [review]
mostly name technicalities

File: src/utils/Firebase/firebase.js (hunk 5 above)
Comment (Lam7150) [reply src/utils/Firebase/firebase.js]: do you mean getDeploymentAccountCustomLessons instead of StudentCustomLessons?

**Review by chyku — APPROVED** [review]
this is kind of insane but why tf do u have so many furniture assigned wordgroups

**Pull request merged**